### ISSUE TITLE
Example object in Schema is always of String type

### ISSUE THREAD

**Issue by dmi3coder** [smallrye/smallrye-open-api#300]
I have a very simple User object where I have `id` field of type integer. I added example with specifying the type
``` java
public class User {

    @Schema(type = SchemaType.INTEGER, example = "1000")
    private Integer id;
... constructor, getters, and setters omitted
}
```
This generates a schema with example as a string
```
    User:
      type: object
      properties:
        id:
          format: int32
          type: integer
          example: "1000"
```
Because of this in e.g Swagger UI when I'm prompted to fill in the example I have string type, users can think that it's mandatory to wrap integer into double-quotes.
While debugging I've found that smallrye-open-api reads value from annotation but doesn't do anything with it. The line below `readAttr()` outputs a String object, which should be of type Integer 
https://github.com/smallrye/smallrye-open-api/blob/143d798943cdf3319790eed1e68d327d64728484/core/src/main/java/io/smallrye/openapi/runtime/io/schema/SchemaFactory.java#L196

<img width="1132" alt="Screen Shot 2020-04-17 at 12 15 51" src="https://user-images.githubusercontent.com/12466860/79555360-21e8fe00-80a8-11ea-9e88-3c6ef84104b1.png">

**Comment by MikeEdgar** [comment]
Hi @dmi3coder - this seems like an issue for the [MicroProfile OpenAPI specification project](https://github.com/eclipse/microprofile-open-api/). If I am understanding you correctly, you are looking for behavior similar to how the `@Extension` annotation is supposed to work, correct? That annotation supports provides a `String` property that must be a JSON object, JSON array, boolean, or number (plus a string).

OpenAPI itself supports any type for the schema example so it sounds like a reasonable suggestion to me. Maybe the change could be to add to `@ExampleObject` a property `boolean parseValue` and support using that annotation from `@Schema`. Please go ahead and open an issue in the specification project. It's possible it could be included in the upcoming 2.0 release.

**Comment by dmi3coder** [comment]
Good day @MikeEdgar, yes, thank you for the detailed description, I submitted pr to micro profile project 👍

**Comment by dmi3coder** [comment]
@MikeEdgar following discussion in https://github.com/eclipse/microprofile-open-api/issues/420, I already investigated how this can be implemented while debugging and I can do a PR with the implementation of this enhancement. Otherwise, I'm happy to leave it for someone else

**Comment by MikeEdgar** [comment]
@dmi3coder +1 to a PR. Thank you!

**Comment by dmi3coder** [comment]
@MikeEdgar I made some progress yesterday, just have a few questions.
1. How do we handle an array type? For now, I decided to just skip it. I can probably take value from the variable itself
https://github.com/dmi3coder/smallrye-open-api/commit/9647fec782880b54d01182057ca967215fa316ad#diff-91c7d8525a1af9ceaec985276adee985R296
2. If we fail to parse a specific field, I assume we should ignore it and return string value. Should we log it somewhere?
